Parser/SwiftParser.swift

```diff
@@ -44,10 +44,21 @@ class SwiftParser {
     var isNextSwitch: Bool = false
     var autoRemoveChar: Bool = false
 
-    init(string: String) {
+    init(string: String, preferences: Preferences? = nil) {
         self.string = string
         self.strIndex = string.startIndex
         self.newlineIndex = string.startIndex
+
+        // Fallback on user-defined preferences
+        guard let preferences = preferences else {
```

Comment:
I think it is better to use `if let` than `guard` because no matter preference is null or not makes no big difference.